components/skeleton/skeleton-mixin.js

```diff
@@ -168,6 +170,55 @@ export const SkeletonMixin = dedupeMixin(superclass => class extends RtlMixin(su
 	constructor() {
 		super();
 		this.skeleton = false;
+		this._skeletonActive = false;
+		this._readyToDisplay = true;
+		this._waitToDisplay = false;
+
+		this._parentSkeleton = new EventSubscriberController(this, 'skeleton', {
+			onSubscribe: this._onSubscribe.bind(this),
+			onUnsubscribe: this._onUnsubscribe.bind(this)
+		});
+	}
+
+	get skeleton() {
+		return this._skeletonActive;
+	}
+
+	set skeleton(val) {
+		const oldVal = this._skeleton;
+		if (oldVal !== val) {
+			this._skeleton = val;
+			this.requestUpdate('skeleton', oldVal);
+			this.requestUpdate('_skeletonActive', oldVal);
+		}
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+		if (changedProperties.has('skeleton')) {
+			this._updateDisplayState();
+			this._parentSkeleton._registryController?.updateSubscribers();
+		}
+	}
+
+	changeDisplay(renderContent) {
+		this._readyToDisplay = renderContent;
+		this._updateDisplayState();
+	}
+
+	_onSubscribe() {
+		this._waitToDisplay = true;
+		this._readyToDisplay = false;
+		this._updateDisplayState();
+	}
+
+	_onUnsubscribe() {
+		this._waitToDisplay = false;
+		this._readyToDisplay = true;
+		this._updateDisplayState();
 	}
 
+	_updateDisplayState() {
+		if (this._waitToDisplay) { this.skeleton = !this._readyToDisplay; }
+	}
```

Comment:
I would have expected this to be something more like:

```suggestion
	_updateDisplayState() {
		if (this._waitToDisplay) {
			this._skeletonActive = !this._readyToDisplay;
		} else {
			this._skeletonActive = this._skeleton;
		}
	}
```

Essentially, if there's no skeleton group in play (`_waitToDisplay` is `false`), then `_skeletonActive` mirrors whatever the consumer sets `skeleton` to. If there is a group (`_waitToDisplay` is `true`), then it mirrors `_readyToDisplay`.